frontend/src/scenes/notebooks/Notebook/Notebook.tsx

```diff
@@ -11,32 +11,46 @@ import { SCRATCHPAD_NOTEBOOK } from '~/models/notebooksModel'
 import { NotebookConflictWarning } from './NotebookConflictWarning'
 import { NotebookLoadingState } from './NotebookLoadingState'
 import { Editor } from './Editor'
-import { EditorFocusPosition } from './utils'
-import { NotebookSidebar } from './NotebookSidebar'
+import { EditorFocusPosition, JSONContent } from './utils'
+import { NotebookColumnLeft } from './NotebookColumnLeft'
 import { ErrorBoundary } from '~/layout/ErrorBoundary'
 import { NotebookHistoryWarning } from './NotebookHistory'
 import { useWhyDidIRender } from 'lib/hooks/useWhyDidIRender'
+import { NotebookColumnRight } from './NotebookColumnRight'
 
-export type NotebookProps = {
-    shortId: string
-    editable?: boolean
+export type NotebookProps = NotebookLogicProps & {
     initialAutofocus?: EditorFocusPosition
+    initialContent?: JSONContent
+    editable?: boolean
 }
 
-export function Notebook({ shortId, editable = false, initialAutofocus = 'start' }: NotebookProps): JSX.Element {
-    const logic = notebookLogic({ shortId })
-    const { notebook, notebookLoading, editor, conflictWarningVisible, isEditable } = useValues(logic)
-    const { duplicateNotebook, loadNotebook, setEditable } = useActions(logic)
+export function Notebook({
+    shortId,
+    mode,
+    editable = true,
+    initialAutofocus = 'start',
+    initialContent,
+}: NotebookProps): JSX.Element {
+    const logicProps: NotebookLogicProps = { shortId, mode }
+    const logic = notebookLogic(logicProps)
+    const { notebook, notebookLoading, editor, conflictWarningVisible, isEditable, isTemplate, notebookMissing } =
+        useValues(logic)
+    const { duplicateNotebook, loadNotebook, setEditable, setLocalContent } = useActions(logic)
     const { isExpanded } = useValues(notebookSettingsLogic)
 
+    useEffect(() => {
+        if (initialContent && mode === 'canvas') {
```

Comment:
Feels strange to have a different code path for initializing the content in a canvas. Not sure how we do it for notebooks but maybe we should combine them into one

Reply:
Agreed - but its trickier than I hoped. initialContent is basically a canvas-only concept atm. Will leave as is and then yeah we should improve later.